### DISCUSSION TITLE
Nuki Hub v9 & (missing/unavailable) Home Assistant entities

### DISCUSSION THREAD

**Discussion by bcutter** [technyon/nuki_hub#506] [Q&A] [answered]
Have these entities been removed with NH v9?

![image](https://github.com/user-attachments/assets/79b4f911-d74a-47b4-9099-870dce726535)

**Comment by iranl** [comment]
No

**Reply by bcutter** [reply]
Which setting is needed in "Access level configuration" for these button entities?

Since I switched from NH 8 to 9 these entities are not available anymore.

**Comment by bcutter** [comment]
So why are they all gone shortly after updating to NH 9?!?

Tried a lot of things like restarting NH, deleting the topics in NH and HA etc. - all those button entities remain unavailable.

Sounds like I need to file an issue @iranl ?

**Reply by iranl** [reply]
Have you enabled the entities in HA after updating to 9.0?

These buttons have always been disabled by default. In 9.0 they might have gotten a new HA unique id which requires re-enabling the buttons.

No need to open an issue as this probably isn't a bug.

**Reply by bcutter** [reply]
Yes they have been (re)enabled after updating. Few hours later they rendered unavailable:

![image](https://github.com/user-attachments/assets/805c1ed3-2644-491b-accc-d985037f13fc)

There are no other hidden entities so HA side seems fine. How to track this down in MQTT?

**Reply by iranl** [reply]
Have you re-enabled homeassistant discovery in NukiHub? Is the home assistant discovery topic set to "homeassistant" in NukiHub?

You can track home assistant discovery using something like MQTT explorer. Just input "query" in the topic filter and it should only show topics with this word in the topic names or values.

**Reply by bcutter** [reply, marked as answer]
I checked hidden entities once again - and indeed, something changed in a way that HA could not reference the existing entities to the newly provided ones. Example:

![grafik](https://github.com/user-attachments/assets/3b5e71b0-2ab5-47b2-ba9f-141fc553e7e2)

I had to delete the unavailable ones and enable the new/hidden ones (using the same entity_id as the deleted ones had by removing the `_2`). Curious how this happened, but fixed for now. Pardon me to have bothered you with this :-)
